### ISSUE TITLE
Bug in type inference after `isinstance(Type[T], type)`

### ISSUE THREAD

**Issue by tamuhey** [microsoft/pyright#1740]
Bug in `isinstance(Type[T], type)` :

```python
def f():
    ty: type[int] = int
    assert isinstance(ty, type)
    _: NoReturn = ty  # Should Error, but No Error (pyright infers `ty: Never`)

# Generic Examle to reproduce the bug:
T = TypeVar("T")
def g(ty: type[T]) -> type[T]:
    assert isinstance(ty, type)
    return ty  # Error: Expression of type "T@g" cannot be assigned to return type "Type[T@g]"
```

pyright version: 1.1.129

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in Pyright 1.1.130, which I just published. It will also be part of the next release of Pylance.
